### ISSUE TITLE
Error: Could not initialize ideviceinstaller; make sure it is installed and works on your system

### ISSUE THREAD

**Issue by kimiwanghao2008** [appium/appium#6340]
## The problem
Error: Could not initialize ideviceinstaller; make sure it is installed and works on your system. Appium server works fine when start up from command line however error like this pop up when start up via object-c code via Xcode.

## Environment
appium : 1.4.8
Xcode : 6.4
mas os : 10.11.1

## Details
 have setup Appium environment 1.4.8 together with Xcode 6.4. It worked well both for simulator and real device. 
I have written a simple shell script with name 'realdevice.sh' to run on mac like this : 
#/bin/bash
'/Applications/Appium.app/Contents/Resources/node/bin/node' '/Applications/Appium.app/Contents/Resources/node_modules/appium/lib/server/main.js' --command-timeout "7200" --debug-log-spacing --platform-version "8.1.1" --platform-name "iOS" --app "io.appium.TestApp" --udid "c94c475b3d59314686d66da932286d4c89f1a38c" --show-ios-log --show-ios-log --device-name "iPhone 6 Plus"
It can work fine directly via typing command 'sh readdevice.sh' and the python script can run normally.

While I wrote a simple macos command line project via Xcode, I wanted to send the command 'sh readdevice.sh' with code 'system("sh readdevice.sh")', appium server can be started. However when running the python script, "ideviceinstaller: "Could not initialize ideviceinstaller"" is singed like this. 

## Link to Appium logs
[debug] Attempting iOS device log capture via libimobiledevice idevicesyslog. Could not capture device         log using libimobiledevice idevicesyslog. Libimobiledevice probably isn't installed
[debug] Attempting iOS device log capture via deviceconsole
[debug] Creating iDevice object with udid c94c475b3d59314686d66da932286d4c89f1a38c
debug] Couldn't find ideviceinstaller, trying built-in at /Applications/Appium.app/Contents/Resources/node_modules/appium/build/libimobiledevice-macosx/ideviceinstaller
Error: Could not initialize ideviceinstaller; make sure it is installed and works on your system
    at [object Object].IOS.getIDeviceObj (/Applications/Appium.app/Contents/Resources/node_modules/appium/lib/devices/ios/ios.js:909:13)
    at [object Object].IOS.installToRealDevice (/Applications/Appium.app/Contents/Resources/node_modules/appium/lib/devices/ios/ios.js:856:32)
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/async/lib/async.js:607:21
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/async/lib/async.js:246:17
    at iterate (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/async/lib/async.js:146:13)
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/async/lib/async.js:157:25
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/async/lib/async.js:248:21
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/async/lib/async.js:612:34
    at [object Object].<anonymous> (/Applications/Appium.app/Contents/Resources/node_modules/appium/lib/devices/ios/ios-crash-log.js:52:5)
    at [object Object].<anonymous> (/Applications/Appium.app/Contents/Resources/node_modules/appium/lib/devices/ios/ios-crash-log.js:27:5)
    at f (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/node_modules/once/once.js:17:25)
    at Glob.<anonymous> (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/glob.js:131:7)
    at Glob.emit (events.js:107:17)
    at Glob._finish (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/glob.js:168:8)
    at done (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/glob.js:157:12)
    at Glob._processReaddir2 (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/glob.js:351:12)
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/glob.js:288:17
    at RES (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/node_modules/inflight/inflight.js:23:14)
    at f (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/node_modules/once/once.js:17:25)
    at Glob._readdirEntries (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/glob.js:480:10)
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/glob/glob.js:457:12
    at FSReqWrap.oncomplete (fs.js:95:15)
[debug] Responding to client with error: {"status":33,"value":{"message":"A new session could not be created. (Original error: Could not initialize ideviceinstaller; make sure it is installed and works on your system)","origValue":"Could not initialize ideviceinstaller; make sure it is installed and works on your system"},"sessionId":null}

**Comment by imurchie** [comment]
Is `idevicesyslog` or `deviceconsole` on  the path that is used when Xcode executes shell scripts?

**Comment by imurchie** [comment]
I cannot open a Word document. Please copy the text of the server logs into a [gist](https://gist.github.com/).

**Comment by tamilRam** [comment]
I have updated the log with debug enabled

**Comment by imurchie** [comment]
Again, I cannot open a Word document.

**Comment by tamilRam** [comment]
i have uploaded the log texts in the gist.

**Comment by imurchie** [comment]
What gist? I don't see anything.

**Comment by kimiwanghao2008** [comment]
I have solved the problem by editing the javascript file. (1) The problem I have met is caused due to that through Xcode or eclipse, environment variables can not be read successfully as expected. In the file /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/node-idevice/main.js, I have edited the function IDevice.prototype._check_cmd = function () by adding where ideviceinstaller is located at .
(2) The problem  tamilRam met may be caused by the reason that the app has already installed on real device which Appium can not detect it and gave the sign 'info: [debug] App is not installed. Will try to install the app.' I have commented "if (err || !installed) {
          logger.debug("App is not installed. Will try to install the app.");
        } " in function IOS.prototype.installToRealDevice = function (cb) in the file '/Applications/Appium.app/Contents/Resources/node_modules/appium/lib/devices/ios '. I am not sure what caused this happen. By doing so, the phenomenon disappeared. You can have a check.
